tool/tsh/tsh.go

```diff
@@ -3132,9 +3136,29 @@ func makeClientForProxy(cf *CLIConf, proxy string, useProfileLogin bool) (*clien
 	// addresses from the proxy (this is what Ping does).
 	if cf.IdentityFileIn != "" {
 		log.Debug("Pinging the proxy to fetch listening addresses for non-web ports.")
-		if _, err := tc.Ping(cf.Context); err != nil {
+		pr, err := tc.Ping(cf.Context)
+		if err != nil {
 			return nil, trace.Wrap(err)
 		}
+		tc.LoadAllCAs = pr.Auth.LoadAllHostCAs
+
+		if tc.LoadAllCAs {
+			sites, err := key.GetClusterNames()
+			if err != nil {
+				return nil, trace.Wrap(err)
+			}
+			hostAuthFunc, err := key.HostKeyCallbackForClusters(cf.InsecureSkipVerify, sites)
+			if err != nil {
+				return nil, trace.Wrap(err)
+			}
+			tc.Config.HostKeyCallback = hostAuthFunc
```

Comment:
Just to make it a little cleaner.
```suggestion
			tc.Config.HostKeyCallback, err = key.HostKeyCallbackForClusters(cf.InsecureSkipVerify, sites)
			if err != nil {
				return nil, trace.Wrap(err)
			}
```